cypress/integration/apps/query_enhancements/dataset_selector.spec.js

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+import {
+  MiscUtils,
+  TestFixtureHandler,
+} from '@opensearch-dashboards-test/opensearch-dashboards-test-library';
+import { PATHS, SECONDARY_ENGINE } from '../../../utils/constants';
+
+const miscUtils = new MiscUtils(cy);
+const testFixtureHandler = new TestFixtureHandler(cy, PATHS.ENGINE);
+
+describe('dataset selector', { scrollBehavior: false }, () => {
+  describe('empty state', () => {
+    it('no index pattern', function () {
+      // Go to the Discover page
+      miscUtils.visitPage(
+        `app/data-explorer/discover#/?_g=(filters:!(),time:(from:'2015-09-19T13:31:44.000Z',to:'2015-09-24T01:31:44.000Z'))`
+      );
+
+      cy.waitForLoaderNewHeader();
```

Comment:
Related to comment above: this can then be rolled into the Discover page visit helper so that no other consumers need to worry about load timing.

Again, no need to address for this PR. Just sharing some thoughts for longterm changes.